### PR TITLE
move overviews to index pages

### LINKED ISSUE #200: move overview pages to index
## Describe the bug

GoQuorum still has separate "overview" pages (e.g. https://consensys.net/docs/goquorum/en/latest/deploy/install/getting-started-overview/) even though index pages are now supported (allowing the navigation folder text to link directly to the index page).

- [ ] Missing content
- [ ] Outdated content
- [ ] Wrong content
- [ ] Confusing or misleading content
- [x] Other

## Change suggestion

Rename all overview pages (e.g. https://consensys.net/docs/goquorum/en/latest/deploy/install/getting-started-overview/) to `index.md` and update navigation accordingly.

### PR DESCRIPTION
**Pull request by alexandratran**
# Pull request checklist

Use the following template to make sure your PR fits the GoQuorum documentation standard.

## Before creating the PR

Make sure that:

- [x] You've read the [contribution guidelines](https://consensys.net/docs/doctools/).
- [x] You've [previewed your changes locally](https://consensys.net/docs/doctools/en/latest/preview/new-system/#preview-locally).

## Describe the change

Rename overview pages as index pages, update navigation and homepage accordingly, and additional minor edits across the index pages.

Fix left padding issues raised by lint codebase.

## Issue fixed

fixes #200 

## Impacted parts

For content changes:

- [x] Documentation content
- [x] Documentation page organization

For tool changes:

- [ ] Build and QA tools (e.g., lint)
- [ ] MkDocs templates
- [x] MkDocs configuration
- [ ] Python dependencies
- [ ] Node dependencies and JavaScript
- [ ] GitHub integration

## After creating your PR and tests have finished

Make sure that:

- [x] You've [fixed any issues](https://consensys.net/docs/doctools/en/latest/contribute/fix-cicd-errors/) raised by the tests.
- [x] You've [previewed your changes on GitHub Pages](https://consensys.net/docs/doctools/en/latest/preview/new-system/#preview-on-github-pages)

### CODE REVIEW

File: mkdocs.navigation.yml [hunk 1]
```diff
@@ -110,7 +110,7 @@ nav:
           - Interact with a deployed contract: tutorials/contracts/calling-contract-functions.md
           - Transfer account funds: tutorials/contracts/account-funds-transfers.md
       - Kubernetes:
-          - Overview: tutorials/kubernetes/overview.md
+          - tutorials/kubernetes/index.md
```
Comment (NicolasMassart) [inline mkdocs.navigation.yml:113]: add a redirect

File: mkdocs.navigation.yml [hunk 2]
```diff
@@ -67,22 +67,22 @@ nav:
...
       - Architecture: concepts/architecture.md
       - Consensus:
-          - Overview: concepts/consensus/overview.md
+          - concepts/consensus/index.md
           - Comparing PoA consensus protocols: concepts/consensus/comparing-poa.md
       - Account management: concepts/account-management.md
       - Free gas network: concepts/free-gas-network.md
       - Privacy:
-          - Overview: concepts/privacy/privacy.md
+          - concepts/privacy/index.md
```
Comment (NicolasMassart) [inline mkdocs.navigation.yml:85]: add a redirect

File: mkdocs.navigation.yml [hunk 3]
```diff
@@ -67,22 +67,22 @@ nav:
...
       - Develop plugins: develop/develop-plugins.md
   - Learn:
       - Blockchain basics:
-          - Introduction: concepts/blockchain-basics/what-is-blockchain.md
+          - concepts/blockchain-basics/index.md
           - Components: concepts/blockchain-basics/blockchain-components.md
       - Architecture: concepts/architecture.md
       - Consensus:
-          - Overview: concepts/consensus/overview.md
+          - concepts/consensus/index.md
```
Comment (NicolasMassart) [inline mkdocs.navigation.yml:80]: add a redirect

File: mkdocs.navigation.yml [hunk 4]
```diff
@@ -67,22 +67,22 @@ nav:
...
       - Manage keys:
-          - Overview: develop/manage-keys/managing-keys.md
+          - develop/manage-keys/index.md
           - Use clef: develop/manage-keys/clef.md
           - Use account plugins: develop/manage-keys/account-plugins.md
       - Develop plugins: develop/develop-plugins.md
   - Learn:
       - Blockchain basics:
-          - Introduction: concepts/blockchain-basics/what-is-blockchain.md
+          - concepts/blockchain-basics/index.md
```
Comment (NicolasMassart) [inline mkdocs.navigation.yml:76]: add a redirect

File: mkdocs.navigation.yml [hunk 5]
```diff
@@ -67,22 +67,22 @@ nav:
       - Client libraries: develop/client-libraries.md
       - Smart contracts and transactions: develop/smart-contracts-transactions.md
       - Manage keys:
-          - Overview: develop/manage-keys/managing-keys.md
+          - develop/manage-keys/index.md
```
Comment (NicolasMassart) [inline mkdocs.navigation.yml:70]: add a redirect

File: mkdocs.navigation.yml [hunk 6]
```diff
@@ -17,7 +17,7 @@ INHERIT: mkdocs.extra.yml # DO NOT MODIFY THIS LINE
 nav:
   - Deploy:
       - Install:
-          - Overview: deploy/install/getting-started-overview.md
+          - deploy/install/index.md
```
Comment (NicolasMassart) [inline mkdocs.navigation.yml:20]: add a redirect

**Review by NicolasMassart — APPROVED** [review]
just redirects, otherwise it's fine for me.

**Review by rolandtyler — APPROVED** [review]
lgtm

**Pull request merged**